doc/refactoring_design.md

```diff
@@ -0,0 +1,481 @@
+
+A bonding curves is a fixed curve that bonds reserved token (i.e., Ether or Ocean token) with a local ERC20 boned token for an associated SA:
+
+<img src="img/refactoring/BC.jpg" width="1000" />
+
+The bonding curves determines the exchange ratio with an analytical formula, which models the price of bonded token as a function of its total supply. Usually price goes up when total supply increases, which indicates strong demand of bonded tokens.
+
+**(1) Curation for SA template**
+
+* Each service agreement template has its own fixed bonding curve along with ERC20 bonded tokens;
+* Whenever the ERC20 bonded token is minted or burnt, the fixed bonding curve is used to calculate the price of bonded tokens:
+	* calculates how many bonded tokens should be minted for a given amount of Ocean tokens;
+	* calculates how many Ocean tokens should be returned when a given amount of bonded tokens are burnt.
+
+**(2) The design structure**:
+
+<img src="img/refactoring/BCstructure.jpg" width="700" />
+
+**(3) Staking and mint new bonded token**:
+<img src="img/refactoring/BCstaking.jpg" width="1100" />
+
+**(4) Unstaking and burn bonded token**:
+<img src="img/refactoring/BCunstaking.jpg" width="1100" />
+
+The example smart contract functions for curation are below:
+
+
+```Solidity
+struct Curation {
+   // fixed bonding curve
+	BondingCurve private bondingCurve;
+	// bonded token	
+	BondedToken private bondedToken;
+}
+
+// each SA template has its own Bonding Curve and Bonded Token
+mapping(bytes32 => Curation) private curations;
+
+// create new curation market for a new service agreement template
+function createCurationMarket(bytes32 template_Id) public returns (bool);
+
+// buy bonded token from a specific bonding curve
+function mintBondedToken(bytes32 template_Id, uint256 numOceanToken) public returns (uint256 numBondedToken);
+
+// sell bonded token into a specific bonding curve
+function burnBondedToken(bytes32 template_Id, uint256 numBondedToken) public returns (uint256 numOceanToken);
+```
+
+The Bonded tokens is fungible ERC20 token as following:
+
+```Solidity
+// transfer tokens 
+function transfer(address _to, uint256 _value) public returns (bool);
+
+// mint new tokens for buyDrops transaction
+function mintTokens() public returns (bool success);
+
+// burn tokens for sellDrops transaction
+function burnTokens() public returns (bool success);
+```
+
+
+### 2.8 Dispute Resolution
+
+Consumer can raise dispute before *TIME-OUT* and *PAYMENT-RELEASED* in the service agreement. 
+
+If dispute exists, the consumer payment is locked and cannot be released before the dispute is resolved.
+
+To resolve the dispute, there are two approaches:
+
+* **Voting based resolution**: it enables expert verification (or arbitration) with permissioned voting on the disputes. 
```

Comment:
This kind of voting based resolution has a problem to disclose data assets or data related services to them right ?

Reply:
I think so. These arbitrators need access to data or service results to vote.